torch/quantization/fx/_equalize.py

```diff
@@ -220,3 +222,245 @@ def node_supports_equalization(node: Node, modules) -> bool:
+            input_eq_obs = modules[node.target]
+            assert(isinstance(input_eq_obs, _InputEqualizationObserver))
+            next_node, weight_eq_obs = get_weight_eq_obs(node, model, modules)
+
+            if next_node is None or weight_eq_obs is None:
+                continue
+
+            assert(isinstance(next_node.target, str))
+            weight_eq_obs(modules[next_node.target].weight)
+
+            # Calculate and set the equalization scale values
+            equalization_scale = calculate_equalization_scale(input_eq_obs, weight_eq_obs)
+            input_eq_obs.set_equalization_scale(equalization_scale)
+            weight_eq_obs.set_equalization_scale(equalization_scale)
+
+            weight_eq_obs_dict[next_node.name] = weight_eq_obs
+
+    return weight_eq_obs_dict
+
+def convert_eq_obs(
+    model: GraphModule,
+    modules: Dict[str, nn.Module],
+    weight_eq_obs_dict: Dict[str, _WeightEqualizationObserver],
+) -> None:
+    """ Removes the input equalization observers and replaces them with mul
+    operators whenever applicable. Updates the input quantization observers with
+    the scaled input min/max values. Scales the weights by the current and next
+    equalization scales.
+
+    Before:
+                                     weight values
+                                          |
+                                      WeightEqObs
+                                          |
+                                     WeightQuantObs
+                                          |
+        x -> InpQuantObs -> InpEqObs -> linear -> OutQuantObs
+
+    After:
+                                              scaled weight values
+                                                      |
+                                                 WeightEqObs
+                                                      |
+       equalization scale                       WeightQuantObs
+              |                                       |
+        x -> mul -> InpQuantObs (scaled min/max) -> linear -> OutQuantObs
+    """
+    for node in model.graph.nodes:
+        if node.op == 'call_module' and isinstance(modules[node.target], _InputEqualizationObserver):
+            inp_quant_obs_node = node.args[0]
+            prev_node = inp_quant_obs_node.args[0]
+
+            # TODO: Possible special handling for connected linear layers
+            # Update the following input quantization observer's min/max values
+            scale_input_node(node, modules)
+
+            # Remove the InputEqualization node and add a mul operator before
+            # the quantization observer node that appears before the equalization node
+            # Before: x -> input_quant_obs -> input_eq_obs -> linear
+            # After: x -> mul -> input_quant_obs -> linear
+
+            # Create a node containing the equalization scale
+            with model.graph.inserting_before(inp_quant_obs_node):
+                name = node.name + '_scale'
```

Comment:
Can we call this `equalization_scale`? We can use this function to generate a unique_name https://github.com/pytorch/pytorch/blob/master/torch/quantization/fx/utils.py#L276